apps/server/src/shared/repo/dashboard/dashboard.repo.integration.spec.ts

```diff
@@ -197,29 +197,51 @@ describe('dashboard repo', () => {
 	});
 
 	describe('getUsersDashboard', () => {
-		it('returns a dashboard', async () => {
-			const result = await repo.getUsersDashboard(new ObjectId().toString());
-			expect(result instanceof DashboardEntity).toEqual(true);
-			expect(result.getGrid().length).toBeGreaterThan(0);
+		describe('when user has no dashboard yet', () => {
+			it('generates an empty dashboard ', async () => {
+				const result = await repo.getUsersDashboard(new ObjectId().toString());
+				expect(result instanceof DashboardEntity).toEqual(true);
+				expect(result.getGrid().length).toEqual(0);
+			});
 		});
 
-		it('always returns the same dashboard for same user', async () => {
-			const userId = new ObjectId().toString();
-			const firstDashboard = await repo.getUsersDashboard(userId);
-			// cant manipulate the dashboard, because the entity doesnt support changes yet
-			const secondDashboard = await repo.getUsersDashboard(userId);
-			expect(firstDashboard.id).toEqual(secondDashboard.id);
-			expect(JSON.stringify(firstDashboard)).toEqual(JSON.stringify(secondDashboard));
-		});
+		describe('when user has a dashboard already', () => {
+			it('should return the existing dashboard', async () => {
+				const user = userFactory.build();
+				const course = courseFactory.build({ students: [user], name: 'Mathe' });
+				await em.persistAndFlush([user, course]);
+				const dashboard = new DashboardEntity(new ObjectId().toString(), {
+					grid: [
+						{
+							pos: { x: 1, y: 3 },
+							gridElement: GridElement.FromSingleReference(course),
+						},
+					],
+					userId: user.id,
+				});
+				await repo.persistAndFlush(dashboard);
+
+				const result = await repo.getUsersDashboard(user.id);
+				expect(result.id).toEqual(dashboard.id);
+			});
 
-		it('always returns different dashboard for different users', async () => {
-			const firstUserId = new ObjectId().toString();
-			const secondUserId = new ObjectId().toString();
+			it('always returns the same dashboard for same user', async () => {
+				const userId = new ObjectId().toString();
+				const firstDashboard = await repo.getUsersDashboard(userId);
 
-			const firstDashboard = await repo.getUsersDashboard(firstUserId);
-			const secondDashboard = await repo.getUsersDashboard(secondUserId);
+				const secondDashboard = await repo.getUsersDashboard(userId);
```

Comment:
shouldn't there be at least one other dashboard for another userId to be sufficient with the claim as for now it only seems necessary condition is met

Reply:
we check the behavior for other users in the test below. Here my main concern is that if no dashboard exists, a new one is created, but if one exists already, it should not create a new one but return the existing one. I guess the testcases are formulated a bit wierd, and the way github shows the change doesnt help...

Reply:
but actually, with the new test I added above, this one seems outdated now. I will remove it